deps: update deps

bl-ue opened the pull request:
- [`glob`](https://www.npmjs.com/package/glob) — `7.1.3` → `7.1.6`
- [`husky`](https://www.npmjs.com/package/husky) — `2.2.0` → `5.2.0`
- [`markdownlint-cli`](https://www.npmjs.com/package/markdownlint-cli) — `0.19.0` → `0.27.1`

bl-ue commented:
We should enable Dependabot on this repo, like we do in the [node client's repo](https://github.com/tldr-pages/tldr-node-client).

owenvoke reviewed (approved):
LGTM! I agree about enabling Dependabot. Feel free to do that if you want (or I can if you'd like?). I'd suggest going with weekly updates to reduce the number of PRs we'd get from it.

bl-ue commented:
Will you please do it? I'm not an org owner (yet), so I can't edit the repository in that type of way.

bl-ue commented:
We should probably do it _after_ this PR is merged though, because it might open duplicate PRs :)

bl-ue commented:
Please see the latest commit.  I ran [`npx husky-4-to-5`](https://github.com/remarkablemark/husky-4-to-5#readme) to convert the old format to the new one.

bl-ue commented:
Hmm...`npm ci` is failing locally as well as in the CI!

bl-ue commented:
Not sure what changed to cause this error, but I fixed it by adding the `name` key to `package.json`.   The full error was:
```
TypeError: Cannot read property 'length' of undefined
    at _incorrectWorkingDirectory (~/.npm/node_modules/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:114:60)
    at ~/.npm/node_modules/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:86:44
    at ~/.npm/node_modules/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:218:12
    at callback (~/.npm/node_modules/lib/node_modules/npm/node_modules/graceful-fs/polyfills.js:295:20)
    at FSReqCallback.oncomplete (fs.js:165:5)
```
the code at that path (`npm-lifecycle/index.js:114:60`):
```javascript
function _incorrectWorkingDirectory (wd, pkg) {
  return wd.lastIndexOf(pkg.name) !== wd.length - pkg.name.length
}
```
after a `console.log(pkg)`, I found it to be the parsed contents of `package.json`. 

So, I added `name`.  It really should've always been there, because `package.json` isn't much of a real `package.json` otherwise.  It also needs `version`, because npm displays the package as `tldr-pages@undefined` 😄, coincides with https://github.com/tldr-pages/tldr/pull/5377#issuecomment-794082345.

mebeim commented:
Is the new `.husky/pre-commit` file required for Husky v5? Just want to make sure.

bl-ue commented:
Yes, it is.  You can see they have one, [here](https://github.com/typicode/husky/tree/371934162db0d7af4c32e89618f9a400ec48e38e/.husky).  Also, I used [this package](https://github.com/remarkablemark/husky-4-to-5#readme) to convert from version 2 to 5.  It created a `.gitignore` and added some items, but it didn't add that, I'd say on purpose.

navarroaxel commented on package.json:
```diff
@@ -1,25 +1,24 @@
-  },
-  "private": true
+  "private": true,
+  "devDependencies": {
+    "husky": "^5.1.3"
+  }
```
I like the `devDeps` under the prod `dependencies`, and the `scripts` above `dependencies`. But this `package.json` has a rare order lol.

bl-ue replied:
No problem, I'll do that.  As a matter of fact though, for some reason it's common to see `devDependencies` at the bottom ¯\\_(ツ)_👍🏻

navarroaxel replied:
I really like the idea of an "official" `package.json` linter/prettier.
Is there something more official than this? https://github.com/cameronhunter/prettier-package-json/blob/HEAD/src/defaultOptions.js

bl-ue replied:
https://npmpackagejsonlint.org/ || https://github.com/tclindner/npm-package-json-lint?

sbrl replied:
@bl-ue I think that's because npm adds new properties to the bottom of `package.json`, and generally speaking `devDependencies` is the last thing that most developers add via an `npm install --save-dev {{package_name}}`

sbrl reviewed (approved):
Looks ok to me

What's `husky`, out of curiosity? Something that our GitHub Actions uses?

navarroaxel commented:
> What's `husky`, out of curiosity? Something that our GitHub Actions uses?

@sbrl an easy wrapper to run hooks like `eslint` before `pre-commit` or `pre-push`, etc. Without deal with bash files in the `.git/hooks` directory. Just config in the `package.json`.

bl-ue commented:
I just updated husky `5.1.3` → `5.2.0`, and I upgraded the lockfile format from v1 (npm 6 and lower) to v2 (npm 7) as it's quickly becoming the norm (and we're already [using it in the linter](https://github.com/tldr-pages/tldr-lint/blob/b070d41be2881c5ede9f65eee3c184bb70526056/package-lock.json#L4)).

navarroaxel commented on package.json:
```diff
@@ -1,25 +1,24 @@
-      "pre-commit": "npm test"
-    }
+    "build-index": "node ./scripts/build-index.js > index.json",
+    "postinstall": "husky install"
```
Oh, this is new for Husky v5.

bl-ue commented:
@owenvoke are we ready to merge?

Waples commented:
guess i'll bump @owenvoke
